modules/service/src/main/scala/lucuma/odb/service/ObservationWorkflowService.scala

```diff
@@ -0,0 +1,461 @@
+        // If there is no instrument in the observation, that will get caught with the generatorValidations
+        optInstr.fold(none.pure){ instr =>
+          session.stream(Statements.CfpInstruments)(cid, chunkSize = 1024)
+            .compile
+            .toList
+            .map(l =>
+              if(l.isEmpty || l.contains(instr)) none
+              else ObservationValidation.callForProposals(Messages.invalidInstrument(instr)).some
+            )
+        }
+      }
+
+      private def obsInfo(oid: Observation.Id): F[ObservationValidationInfo] =
+        session.unique(Statements.ObservationValidationInfo)(oid)
+
+      private def itcValidations(
+        info: ObservationValidationInfo,
+        itcClient: ItcClient[F], 
+        params: GeneratorParams
+      )(using Transaction[F]): F[ObservationValidationMap] =
+        itcService(itcClient).selectOne(info.pid, info.oid, params).map:
+          // N.B. there will soon be more cases here
+          case Some(_) => ObservationValidationMap.empty
+          case None    => ObservationValidationMap.singleton(ObservationValidation.itc("ITC results are not present."))
+
+      private def validateScienceBand(oid: Observation.Id): F[ObservationValidationMap] =
+        session
+          .option(Statements.SelectInvalidBand)(oid)
+          .map { invalidBand =>
+            val m = ObservationValidationMap.empty
+            invalidBand.fold(m)(b => m.add(ObservationValidation.configuration(Messages.invalidScienceBand(b))))
+          }
+
+      private def validateConfiguration(oid: Observation.Id)(using Transaction[F]): F[ObservationValidationMap] =
+        configurationService.selectRequests(oid).map: r =>
+          val m = ObservationValidationMap.empty
+          r.toOption match
+            case None => m.add(ObservationValidation.configuration(Messages.ConfigurationRequest.Unavailable))
+            case Some(Nil) => m.add(ObservationValidation.configuration(Messages.ConfigurationRequest.NotRequested))
+            case Some(lst) if lst.exists(_.status === ConfigurationRequestStatus.Approved) => m
+            case Some(lst) if lst.forall(_.status === ConfigurationRequestStatus.Denied) => m.add(ObservationValidation.configuration(Messages.ConfigurationRequest.Denied))
+            case _ => m.add(ObservationValidation.configuration(Messages.ConfigurationRequest.Pending))
+
+      // Construct some finer-grained types to make it harder to do something dumb in the status computation.
+      import ObservationWorkflowState.*
+      type UserState       = Inactive.type  | Ready.type
+      type ExecutionState  = Ongoing.type   | Completed.type
+      type ValidationState = Undefined.type | Unapproved.type | Defined.type
+
+      private def executionState(info: ObservationValidationInfo, commitHash: CommitHash, itcClient: ItcClient[F])(using Enums): F[Option[ExecutionState]] =
```

Comment:
This is new and unused. It gets the newly-available `ExecutionStatus` and translates the result to an `ObservationWorkflowState`.